tools/cudaq-qpud/nvqc_proxy.py

```diff
@@ -194,24 +192,26 @@ def do_POST(self):
                     save_dir = os.getcwd()
                     os.chdir(pathlib.Path(temp_file.name).parent)
                     cmd_list = build_command_list(temp_file.name)
-                    cmd_result = subprocess.run(cmd_list,
-                                                capture_output=False,
-                                                text=True)
-                    if cmd_result.returncode == 0:
+                    try:
+                        subprocess.run(cmd_list,
+                                       timeout=WATCHDOG_TIMEOUT_SEC,
+                                       capture_output=False,
+                                       text=True)
+
                         with open(temp_file.name, 'rb') as fp:
                             result = json.load(fp)
-                    else:
-                        if cmd_result.returncode == 124:
-                            error_message = "Timeout occurred during execution."
-                        else:
-                            error_message = f'Return code: {cmd_result.returncode}\n' + \
-                                            f'{cmd_result.stdout}\n' + \
-                                            f'{cmd_result.stderr}\n'
+
+                        if 'errorMessage' in result:
+                            result = {
+                                'status':
+                                    'json_request_runner.py returned an error',
+                                'errorMessage':
+                                    result['errorMessage']
+                            }
+                    except subprocess.TimeoutExpired as e:
                         result = {
-                            'status':
-                                'json_request_runner.py returned an error',
-                            'errorMessage':
-                                error_message
+                            'status': 'Timeout occurred',
+                            'errorMessage': 'Timeout occurred during execution'
```

Comment:
Recommend switching this back to inspecting the `cmd_result` in order to work with running `json_request_runner.py` as `nobody`.